project-koku/koku
# [Cost-2026]order_by date cannot sort on a day without data

Fixes [COST-2026](https://issues.redhat.com/browse/COST-2026)

Changes proposed in this PR:
Now filtering on a date without data does not remove all valid dates with data from the query. 

Testing Instructions:
1. Spin up docker, make test customer data and load it. 
2. Since Only two months of data are generated we will access a date that is out of range and has no data. 
3. Hit this endpoint http://localhost:8000/api/cost-management/v1/reports/gcp/costs/?filter[limit]=10&filter[offset]=0&group_by[gcp_project]=*&order_by[cost]=desc&order_by[date]=2022-04-30&end_date=2022-06-02&start_date=2022-04-30
4. Expected to see that 04-30 has no data but all other valid dates have data in the given range from 04-30 to 06-02

---
Additional Context:
Please reach out with any questions!
